### PR TITLE
Add GMT+12 MOJURO TimeZone to ReportStream

### PR DESCRIPTION
**Pull request by oslynn**
This PR adds MOJURO timezone to reportStream since Marshall Island needs GMT+ 12 (MAJURO) timezone to be added to their message if we receive a message that has a date filed with a timezone. However, ReportStream doesn't support or allow to setting the timezone, yet.

Test Steps:
1. git pull
2. in prime-reportstream directory run:  
   ./gradlew prime-router:test --tests "gov.cdc.prime.router.ReceiverTests.test MAJURO receiver timezone" <Enter>

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`?
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue
- https://app.zenhub.com/workspaces/engagement-6166edbd409257001e09f1b2/issues/gh/cdcgov/prime-reportstream/16825

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: prime-router/src/test/kotlin/ReceiverTests.kt [hunk 1]
```diff
@@ -273,4 +274,11 @@ internal class ReceiverTests {
         val receiver = Receiver("elr", "IGNORE", Topic.COVID_19, CustomerStatus.INACTIVE, translatorConfig)
         assertThat(receiver.transportType.type).isEqualTo("NULL")
     }
+
+    @Test
+    fun `test MOJURO receiver timezone`() {
```
Comment (the-andrew) [inline prime-router/src/test/kotlin/ReceiverTests.kt:279]: Minor: test name typo.

Also, why is it testing `isNotEqualTo` if it's creating a time in MHT and comparing it to what appears to be the same datetime?

Comment (oslynn) [reply prime-router/src/test/kotlin/ReceiverTests.kt:279]: Thank you very much, Andrew.

File: prime-router/src/test/kotlin/ReceiverTests.kt [hunk 2]
```diff
@@ -273,4 +274,11 @@ internal class ReceiverTests {
         val receiver = Receiver("elr", "IGNORE", Topic.COVID_19, CustomerStatus.INACTIVE, translatorConfig)
         assertThat(receiver.transportType.type).isEqualTo("NULL")
     }
+
+    @Test
+    fun `test MAJURO receiver timezone`() {
+        val shouldWork1 =
+            ZonedDateTime.of(2024, 12, 20, 0, 0, 0, 0, ZoneId.of(USTimeZone.MAJURO.zoneId)).toOffsetDateTime()
+        assertThat(shouldWork1).isNotEqualTo("2024-12-20T00:00+12:00")
+    }
```
Comment (victor-chaparro) [inline prime-router/src/test/kotlin/ReceiverTests.kt:283]: What is this test actually testing? Is testing that it works or that it doesn't work?

Comment (oslynn) [reply prime-router/src/test/kotlin/ReceiverTests.kt:283]: Test to see, it is working.

Comment (victor-chaparro) [reply prime-router/src/test/kotlin/ReceiverTests.kt:283]: Shouldn't it be testing that the date gets converted to MAJURO than?

Comment (oslynn) [reply prime-router/src/test/kotlin/ReceiverTests.kt:283]: Oh yes, are you correct.  I will test the date get converted to MAJURO as well.  Thanks.

Comment (oslynn) [reply prime-router/src/test/kotlin/ReceiverTests.kt:283]: Added the test the date get converted to MAJURO.  Done.

**Review by victor-chaparro — APPROVED** [review]

**Pull request merged**